frontend/app/src/next/components/ui/code/snippet.tsx

```diff
@@ -0,0 +1,59 @@
+import { useMemo } from 'react';
+import { CodeBlock } from './code-block';
+import { Snippet as SnippetType } from '@/next/lib/docs/snips';
+interface SnippetProps {
+  src: SnippetType;
+  block?: keyof SnippetType['blocks'] | 'ALL';
+}
+
+const languageMap = {
+  typescript: 'ts',
+  python: 'py',
+  go: 'go',
+  unknown: 'txt',
+};
+
+// This is a server component that will be rendered at build time
```

Comment:
do we have server components?

Reply:
its statically rendered at build time, maybe its not technically a server component